service/dap/server.go

```diff
@@ -518,42 +565,68 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 
 	s.setLaunchAttachArgs(request)
 
-	var targetArgs []string
-	args, ok := request.Arguments["args"]
-	if ok {
-		argsParsed, ok := args.([]interface{})
-		if !ok {
-			s.sendErrorResponse(request.Request,
-				FailedToLaunch, "Failed to launch",
-				fmt.Sprintf("'args' attribute '%v' in debug configuration is not an array.", args))
-			return
-		}
-		for _, arg := range argsParsed {
-			argParsed, ok := arg.(string)
-			if !ok {
-				s.sendErrorResponse(request.Request,
-					FailedToLaunch, "Failed to launch",
-					fmt.Sprintf("value '%v' in 'args' attribute in debug configuration is not a string.", arg))
-				return
-			}
-			targetArgs = append(targetArgs, argParsed)
-		}
+	targetArgs, ok := s.parseTargetArgs(request, ok)
```

Comment:
What's the intent behind passing ok to parseTargetArgs? It doesn't use it. Also parseTargetArgs is only called here, AFAICT, so I think it should be left inline.

Reply:
My bad, it slipped during refactoring (the method extractor sometimes tends to do this). Fixed